dev/com.ibm.ws.jpa_spec10_query_fat/fat/src/com/ibm/ws/jpa/spec10/query/jpql/TestQueryJPQL_Web.java

```diff
@@ -0,0 +1,183 @@
+import componenttest.custom.junit.runner.Mode;
+import componenttest.custom.junit.runner.Mode.TestMode;
+import componenttest.topology.impl.LibertyServer;
+import componenttest.topology.utils.PrivHelper;
+
+@RunWith(FATRunner.class)
+@Mode(TestMode.FULL)
+public class TestQueryJPQL_Web extends JPAFATServletClient {
+    private final static String queryAppFolder = "jpql";
+    private final static String RESOURCE_ROOT = "test-applications/query/" + queryAppFolder + "/";
+    private final static String appName = "jpql";
+    private final static String appNameEar = appName + ".ear";
+
+    private final static Set<String> dropSet = new HashSet<String>();
+    private final static Set<String> createSet = new HashSet<String>();
+    private final static Set<String> populateSet = new HashSet<String>();
+
+    private static long timestart = 0;
+
+    static {
+        dropSet.add("JPA_QUERY_DROP_${dbvendor}.ddl");
+        createSet.add("JPA_QUERY_CREATE_${dbvendor}.ddl");
+        populateSet.add("JPA_QUERY_POPULATE_${dbvendor}.ddl");
+    }
+
+    @Server("JPAQueryServer")
+    @TestServlets({
+                    @TestServlet(servlet = TestJPQLServlet.class, path = "jpql" + "/" + "TestJPQLServlet"),
+
+    })
+    public static LibertyServer server;
+
+    @BeforeClass
+    public static void setUp() throws Exception {
+        PrivHelper.generateCustomPolicy(server, FATSuite.JAXB_PERMS);
+        bannerStart(TestQueryJPQL_Web.class);
+        timestart = System.currentTimeMillis();
+
+        int appStartTimeout = server.getAppStartTimeout();
+        if (appStartTimeout < (120 * 1000)) {
+            server.setAppStartTimeout(120 * 1000);
+        }
+
+        int configUpdateTimeout = server.getConfigUpdateTimeout();
+        if (configUpdateTimeout < (120 * 1000)) {
+            server.setConfigUpdateTimeout(120 * 1000);
+        }
+
+        server.startServer();
+
+        setupDatabaseApplication(server, RESOURCE_ROOT + "ddl/");
+
+        final Set<String> ddlSet = new HashSet<String>();
+
+        System.out.println("TestQueryJPQL_Web Setting up database tables...");
+
+        ddlSet.clear();
+        for (String ddlName : dropSet) {
+            ddlSet.add(ddlName.replace("${dbvendor}", getDbVendor().name()));
+        }
+        executeDDL(server, ddlSet, true);
+
+        ddlSet.clear();
+        for (String ddlName : createSet) {
+            ddlSet.add(ddlName.replace("${dbvendor}", getDbVendor().name()));
+        }
+        executeDDL(server, ddlSet, false);
+
+        ddlSet.clear();
+        for (String ddlName : populateSet) {
+            ddlSet.add(ddlName.replace("${dbvendor}", getDbVendor().name()));
+        }
+        executeDDL(server, ddlSet, false);
+
+        setupTestApplication();
+    }
+
+    private static void setupTestApplication() throws Exception {
+        WebArchive webApp = ShrinkWrap.create(WebArchive.class, appName + ".war");
+        webApp.addPackages(true, "com.ibm.ws.jpa.query.sqlcapture");
```

Comment:
the `true` boolean here means automatically adding sub-packages. Would it be simpler to just do:
`webApp.addPackages(true, "com.ibm.ws.jpa.query");`

Reply:
Ah, good point there.